src/main/java/com/bringframework/context/ApplicationContext.java

```diff
@@ -0,0 +1,44 @@
+package com.bringframework.context;
+
+import com.bringframework.exceptions.NoSuchBeanException;
+import com.bringframework.exceptions.NoUniqueBeanException;
+import java.util.Map;
+
+/**
+ * Interface for an Inversion of Control (IoC) container context. An IoC container is responsible
+ * for instantiating, configuring, and assembling objects known as beans, as well as managing their
+ * lifecycle.
+ */
+public interface ApplicationContext {
+
+  /**
+   * Retrieves a bean of the specified type from the container.
+   *
+   * @param beanType the type of the bean to retrieve
+   * @param <T>      the generic type of the bean
+   * @return the bean of the specified type
+   * @throws NoSuchBeanException   if no bean of the specified type is found
+   * @throws NoUniqueBeanException if multiple beans of the specified type are found
+   */
+  <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException;
```

Comment:
If I understand it right, as suggested by Taras Boychuk last home review session:  remove throws declarations from methods as they 1) are runtime exceptions 2) overwhelm usage of an public api.

Reply:
Removed.